sdrf_pipelines/maxquant/maxquant.py

```diff
@@ -2013,16 +2013,18 @@ def maxquant_convert(
             Variable_list.extend(j["mods"][1].split(","))
             fixedM_list = list(set(fixedM_list))
             Variable_list = list(set(Variable_list))
-            for F in fixedM_list:
-                string = doc.createElement("string")
-                string.appendChild(doc.createTextNode(F))
-                fixedModifications.appendChild(string)
-            for V in Variable_list:
-                if "Lys8" == V or "Lys6" == V or "Lys4" == V or "Arg10" == V or "Arg6" == V:
-                    continue
-                string = doc.createElement("string")
-                string.appendChild(doc.createTextNode(V))
-                variableModifications.appendChild(string)
+            if len(fixedM_list) > 0: 
+                for F in fixedM_list:
+                    string = doc.createElement("string")
+                    string.appendChild(doc.createTextNode(F))
+                    fixedModifications.appendChild(string)
+            if len(Variable_list) > 0:
+                for V in Variable_list:
```

Comment:
How does the if help? If e.g. fixedM_list has no elements, the code inside the loop will not be executed. Same for `Variable_list`.

Reply:
OK, the if clause makes more sense now, but now you'll get an IndexError if either `Variable_list` or `fixedM_list` are empty lists. (But I am not sure if this can happen or not.)

Reply:
At least in my current tests it does not happen. Maxquant throws an error during the alignment (after hours of running) if there is a `<string></string>`, which is why I added this.

Reply:
It does not seem to happen. I observe at least one element which then can be empty.

Reply:
hmmm. I think I know what's happening.

In line 2012 `j["mods"][0].split(",")` will actually be a list with one element which is an empty string `['']` if `j["mods"][0]` is an empty string.

I would suggest to put `if j["mods"][0]:` before the split, to only run it if mods are supplied. (implicit falsey empty string)

Reply:
Done. But strangely I cannot run the tool anymore. Independently of the command, I get
`Error: No such command`
when running e.g. 
`parse_sdrf convert_maxquant -s FILE`

Same for `validate_sdrf` and others.

Reply:
oh, this is not exactly what I had in mind. This change will not work as expected. Here's a more verbose suggestion:

```
            def parse_mods(mods):
                mods_list = []
                if mods != "":
                    mods_list.extend(mods.split(","))
                return list(set(mods_list))

            fixedM_list = parse_mods(j["mods"][0])
            Variable_list = parse_mods(j["mods"][1])
            for F in fixedM_list:
                string = doc.createElement("string")
                string.appendChild(doc.createTextNode(F))
                fixedModifications.appendChild(string)
            for V in Variable_list:
```

Reply:
I normally use `pip install [-e] .` in a virtual environment  to have the command installed. Without knowledge of the setup, it is near impossible to advise. I would be surprised if the lack of the command is related to this change.

Reply:
Indeed much better. My python skills are not the most advanced, so I apologize.